fvm/src/kernel/default.rs

```diff
@@ -125,6 +125,8 @@ where
             .unwrap_or(false);
 
         if !is_account {
+            // TODO: this is wrong. Maybe some InvalidActor type?
+            // The argument is syntactically correct, but semantically wrong.
```

Comment:
I don't think adding such as specialised error number is useful. It's really just an invalid argument. We might be getting caught up in the nuance between `invalid` and `illegal`. I take this error as meaning "this argument was invalid" without any implication of the cause (invalid syntax, or invalid value). We may want to rename `IllegalArgument` to `InvalidArgument` to open the definition more. (The [spec](https://github.com/filecoin-project/fvm-specs/blob/main/07-errors.md) mixes invalid and illegal).

Reply:
I'd like to be able to distinguish between a _bug_ in the actor and a "normal" error. In most cases, `IllegalArgument` means that there's a bug in the actor itself.

But this is hardly the only borderline case. Most of the proof code will shrug and say "eh, illegal argument?".